src/main/java/edu/ie3/datamodel/io/source/couchbase/CouchbaseWeatherSource.java

```diff
@@ -0,0 +1,171 @@
+  }
+
+  @Override
+  public Map<Point, IndividualTimeSeries<WeatherValue>> getWeather(
+      ClosedInterval<ZonedDateTime> timeInterval) {
+    logger.warn(
+        "By not providing coordinates you are forcing couchbase to check all possible coordinates one by one."
+            + " This is not very performant. Please consider providing specific coordinates instead.");
+    return getWeather(timeInterval, coordinateSource.getAllCoordinates());
+  }
+
+  @Override
+  public Map<Point, IndividualTimeSeries<WeatherValue>> getWeather(
+      ClosedInterval<ZonedDateTime> timeInterval, Collection<Point> coordinates) {
+    HashMap<Point, IndividualTimeSeries<WeatherValue>> coordinateToTimeSeries = new HashMap<>();
+    for (Point coordinate : coordinates) {
+      String query = createQueryStringForIntervalAndCoordinate(timeInterval, coordinate);
+      CompletableFuture<QueryResult> futureResult = connector.query(query);
+      QueryResult queryResult = futureResult.join();
+      List<JsonObject> jsonWeatherInputs = Collections.emptyList();
+      try {
+        jsonWeatherInputs = queryResult.rowsAsObject();
+      } catch (DecodingFailureException ex) {
+        logger.error(ex);
+      }
+      if (jsonWeatherInputs != null && !jsonWeatherInputs.isEmpty()) {
+        Set<TimeBasedValue<WeatherValue>> weatherInputs =
+            jsonWeatherInputs.stream()
+                .map(this::toTimeBasedWeatherValue)
+                .flatMap(o -> o.map(Stream::of).orElseGet(Stream::empty))
+                .collect(Collectors.toSet());
+        IndividualTimeSeries<WeatherValue> weatherTimeSeries =
+            new IndividualTimeSeries<>(null, weatherInputs);
+        coordinateToTimeSeries.put(coordinate, weatherTimeSeries);
+      }
+    }
+    return coordinateToTimeSeries;
+  }
+
+  @Override
+  public Optional<TimeBasedValue<WeatherValue>> getWeather(ZonedDateTime date, Point coordinate) {
+    try {
+      CompletableFuture<GetResult> futureResult =
+          connector.get(generateWeatherKey(date, coordinate));
+      GetResult getResult = futureResult.join();
+      JsonObject jsonWeatherInput = getResult.contentAsObject();
+      return toTimeBasedWeatherValue(jsonWeatherInput);
+    } catch (DecodingFailureException ex) {
+      logger.error(ex);
+      return Optional.empty();
+    } catch (DocumentNotFoundException ex) {
+      return Optional.empty();
+    } catch (CompletionException ex) {
+      if (ex.getCause() instanceof DocumentNotFoundException) return Optional.empty();
+      else throw ex;
+    }
+  }
+
+  /**
+   * Generates a key for weather documents with the pattern: {@code
+   * weather::<coordinate_id>::<time>}
+   *
+   * @param time the timestamp for the weather data
+   * @param coordinate the coordinate of the weather data
+   * @return a weather document key
+   */
+  public String generateWeatherKey(ZonedDateTime time, Point coordinate) {
+    String key = "weather::";
```

Comment:
is this always the same? Otherwise pls prevent hardcoding it, but make it a final field and add it to a parametrized constructor